refactor(x/ecocredit): rename ecocredit core functionality to base

ryanchristo opened the pull request:
<!--
The default pull request template is for types feat, fix, or refactor.
For other templates, add one of the following parameters to the url:
- template=docs.md
- template=other.md
-->

## Description

Ref: #1196, #1429

<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review. -->

Finish renaming core to base following #1429.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

technicallyty reviewed (approved):
pre-approving

found a few lingerings of "core":
- variable name `coreQuerier` should now be `baseQuerier` https://github.com/regen-network/regen-ledger/blob/ryan/rename-core-to-base/x/ecocredit/module/module.go#L269
- comment `ecocredit core` should now be `ecocredit base` https://github.com/regen-network/regen-ledger/blob/ryan/rename-core-to-base/x/ecocredit/genesis/genesis.go#L250
- varable name `coreQryClient` should now be `baseQryClient` https://github.com/regen-network/regen-ledger/blob/ryan/rename-core-to-base/x/ecocredit/marketplace/simulation/operations.go#L271
- theres a few gas iteration descriptions that still path to "core" in msg server functions create_class, retire, send, update_class_issuers, marketplace.sell
- all our codecs still register the types with regen.core, but i am unsure if that should change/could cause problems? 🤔 do you have any insight on that? 

not sure if all of these should be addressed here, but figured i'd point em out just in case 👍🏻

ryanchristo commented:
> all our codecs still register the types with regen.core, but i am unsure if that should change/could cause problems? thinking do you have any insight on that?

As far as I can tell, this should be ok to update. Might require a change for app developers so added a changelog entry for this specifically. This should probably not have been `regen.core/...` in the first place but rather `regen/...` to align with the type urls and proto package structure. I removed `core` from gas iterations rather than updating them to `base` for the same reason (and fixed the reference in the marketplace sell method).

aleem1314 reviewed (approved):
lgtm
